transaction_creation/marketplace_transaction/common.py

```diff
@@ -0,0 +1,110 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+
+import hashlib
+from uuid import uuid4
+
+from sawtooth_rest_api.protobuf import batch_pb2
+from sawtooth_rest_api.protobuf import transaction_pb2
+
+from sawtooth_signing import create_context
+
+from marketplace_addressing import addresser
+
+
+context = create_context('secp256k1')
+
+
+def wrap_payload_in_txn_batch(txn_key, payload, header, batch_key):
+    """Takes the serialized RBACPayload and creates a batch_list, batch
+    signature tuple.
+    Args:
+        txn_key (Key): The txn signer's public/private key pair.
+        payload (bytes): The serialized RBACPayload.
+        header (bytes): The serialized TransactionHeader.
+        batch_key (Key): The batch signer's public/private key pair.
+    Returns:
+        tuple
+            The zeroth element is a BatchList, and the first element is
+            the batch header_signature.
+    """
+
+    transaction = transaction_pb2.Transaction(
+        payload=payload,
+        header=header,
+        header_signature=context.sign(header, txn_key.private_key))
+
+    batch_header = batch_pb2.BatchHeader(
+        signer_public_key=batch_key.public_key,
+        transaction_ids=[transaction.header_signature]).SerializeToString()
+
+    batch = batch_pb2.Batch(
+        header=batch_header,
+        header_signature=context.sign(batch_header, batch_key.private_key),
+        transactions=[transaction])
+
+    batch_list = batch_pb2.BatchList(
+        batches=[batch])
+    return batch_list, batch.header_signature
+
+
+def make_header_and_batch(payload, inputs, outputs, txn_key, batch_key):
+
+    header = make_header(
+        inputs=inputs,
+        outputs=outputs,
+        payload_sha512=hashlib.sha512(
+            payload.SerializeToString()).hexdigest(),
+        signer_pubkey=txn_key.public_key,
+        batcher_pubkey=batch_key.public_key)
+
+    return wrap_payload_in_txn_batch(
+        txn_key=txn_key,
+        payload=payload.SerializeToString(),
+        header=header.SerializeToString(),
+        batch_key=batch_key)
+
+
+def make_header(inputs,
+                outputs,
+                payload_sha512,
+                signer_pubkey,
+                batcher_pubkey):
+    header = transaction_pb2.TransactionHeader(
+        inputs=inputs,
+        outputs=outputs,
+        batcher_public_key=batcher_pubkey,
+        dependencies=[],
+        family_name=addresser.FAMILY_NAME,
+        family_version='1.0',
+        nonce=uuid4().hex,
+        signer_public_key=signer_pubkey,
+        payload_sha512=payload_sha512)
+    return header
+
+
+class Key(object):
```

Comment:
You don't need to create this Key class.  You should use the sawtooth_signing.Signer class instead.  It encapsulates the context and the private key.  You can access the public key via `signer.get_public_key().as_hex()` for use in the headers.  See the updated [developer's guide](https://sawtooth.hyperledger.org/docs/core/releases/1.0.0rc3/_autogen/sdk_submit_tutorial_python.html) on this.

Reply:
Thanks @peterschwarz . I will look into that.